packages/subproviders/src/subproviders/mnemonic_wallet.ts

```diff
@@ -0,0 +1,138 @@
+import { assert } from '@0xproject/assert';
+import { addressUtils } from '@0xproject/utils';
+import * as bip39 from 'bip39';
+import ethUtil = require('ethereumjs-util');
+import HDNode = require('hdkey');
+import * as _ from 'lodash';
+
+import { DerivedHDKey, PartialTxParams, WalletSubproviderErrors } from '../types';
+import { walletUtils } from '../utils/wallet_utils';
+
+import { BaseWalletSubprovider } from './base_wallet_subprovider';
+import { PrivateKeyWalletSubprovider } from './private_key_wallet';
+
+const DEFAULT_DERIVATION_PATH = `44'/60'/0'/0`;
+
+/**
+ * This class implements the [web3-provider-engine](https://github.com/MetaMask/provider-engine) subprovider interface.
+ * This subprovider intercepts all account related RPC requests (e.g message/transaction signing, etc...) and handles
+ * all requests with accounts derived from the supplied mnemonic.
+ */
+export class MnemonicWalletSubprovider extends BaseWalletSubprovider {
+    private _addressSearchLimit: number;
+    private _derivationBasePath: string;
+    private _derivedKey: DerivedHDKey;
+    /**
+     * Instantiates a MnemonicWalletSubprovider. Defaults to derivationPath set to `44'/60'/0'/0`.
+     * This is the default in TestRPC/Ganache, this can be overridden if desired.
+     * @param mnemonic The mnemonic seed
+     * @param derivationBasePath The derivation path, defaults to `44'/60'/0'/0`
+     * @param addressSearchLimit The limit on address search attempts before raising `WalletSubproviderErrors.AddressNotFound`
+     * @return MnemonicWalletSubprovider instance
+     */
+    constructor(
+        mnemonic: string,
+        derivationBasePath: string = DEFAULT_DERIVATION_PATH,
+        addressSearchLimit: number = walletUtils.DEFAULT_ADDRESS_SEARCH_LIMIT,
```

Comment:
The two above should be nested within a config object. Otherwise, I **must** set the first in order to set the second. This is cumbersome.

Also make `DEFAULT_ADDRESS_SEARCH_LIMIT` local to this file.